mlflow/utils/autologging_utils/client.py

```diff
@@ -0,0 +1,325 @@
+        )
+        params_arr = [Param(key, str(value)) for key, value in params.items()]
+        self._get_pending_operations(run_id).enqueue(params=params_arr)
+
+    def log_metrics(
+        self,
+        run_id: Union[str, PendingRunId],
+        metrics: Dict[str, float],
+        step: Optional[int] = None,
+    ) -> None:
+        """
+        Enqueues a collection of Metrics to be logged to the run specified by `run_id` at the
+        step specified by `step`.
+        """
+        metrics = _truncate_dict(metrics, max_key_length=MAX_ENTITY_KEY_LENGTH)
+        timestamp = int(time.time() * 1000)
+        metrics_arr = [Metric(key, value, timestamp, step or 0) for key, value in metrics.items()]
+        self._get_pending_operations(run_id).enqueue(metrics=metrics_arr)
+
+    def set_tags(self, run_id: Union[str, PendingRunId], tags: Dict[str, Any]) -> None:
+        """
+        Enqueues a collection of Tags to be logged to the run specified by `run_id`.
+        """
+        tags = _truncate_dict(
+            tags, max_key_length=MAX_ENTITY_KEY_LENGTH, max_value_length=MAX_TAG_VAL_LENGTH
+        )
+        tags_arr = [RunTag(key, str(value)) for key, value in tags.items()]
+        self._get_pending_operations(run_id).enqueue(tags=tags_arr)
+
+    def flush(self, synchronous=True):
+        """
+        Flushes all queued run operations, resulting in the creation or mutation of runs
+        and run data.
+
+        :param synchronous: If `True`, run operations are performed synchronously, and a
+                            `RunOperations` result object is only returned once all operations
+                            are complete. If `False`, run operations are performed asynchronously,
+                            and an `RunOperations` object is returned that represents the ongoing
+                            run operations.
+        :return: A `RunOperations` instance representing the flushed operations. These operations
+                 are already complete if `synchronous` is `True`. If `synchronous` is `False`, these
+                 operations may still be inflight. Operation completion can be synchronously waited
+                 on via `RunOperations.await_completion()`.
+        """
+        logging_futures = []
+        for pending_operations in self._pending_ops_by_run_id.values():
+            future = self._thread_pool.submit(
+                self._flush_pending_operations, pending_operations=pending_operations,
+            )
+            logging_futures.append(future)
+        self._pending_ops_by_run_id = {}
+
+        logging_operations = RunOperations(logging_futures)
+        if synchronous:
+            logging_operations.await_completion()
+        return logging_operations
+
+    def _get_pending_operations(self, run_id):
+        """
+        :return: A `_PendingRunOperations` containing all pending operations for the specified
```

Comment:
```suggestion
        :return: A `_PendingRunOperations` containing all pending operations for the specified `run_id`
```

Is `run_id` missing?

Reply:
It definitely was. I've fixed this. Thanks Haru!